src/sys/settings.c

```diff
@@ -87,50 +88,17 @@ void applyMenuSettings(void) {
 }
 
 void settingsSave(s32 profile) {
-    u16 *checksum = &settingsStore.header.dataChecksum;
-    *checksum = settingsChecksumCompute(&settingsStore);
-
-    // read in save file in the same slot as the profile
-    char *start = malloc(SETTINGS_SAVE_FILE_SIZE + sizeof(settingsStore));
-    pm_fioReadFlash(profile, start, SETTINGS_SAVE_FILE_SIZE);
-
-    // append settings data to the end and save to file
-    char *offset = start + SETTINGS_SAVE_FILE_SIZE;
-    memcpy(offset, &settingsStore, sizeof(settingsStore));
-    pm_fioWriteFlash(profile, start, SETTINGS_SAVE_FILE_SIZE + sizeof(settingsStore));
-    free(start);
-}
-
-static bool saveFileExists(void) {
-    char *file = malloc(SETTINGS_SAVE_FILE_SIZE);
-    for (s32 i = 0; i < 8; i++) {
-        pm_fioReadFlash(i, file, SETTINGS_SAVE_FILE_SIZE);
-        if (pm_fioValidateFileChecksum(file)) {
-            free(file);
-            return TRUE;
-        }
-    }
-    free(file);
-    return FALSE;
+    pm_fioReadFlash(SETTINGS_FIO_PAGE, &settingsBuffer, sizeof(settingsBuffer));
```

Comment:
This seems to overwrite the first 4 bytes of settingsStore. Is pm_fioReadFlash reading in blocks of 128 bytes? might be an alignment issue

Reply:
It's possible I'm just having a moment here, but are you sure? I'm watching it in the memory viewer and the only thing in settingsStore I see changing is the checksum.